### ISSUE TITLE
Ignore trailing whitespaces on searches

### ISSUE THREAD

**Issue by j-ed** [nextcloud/android#8262]
### Actual behavior
If you enter a part of keyword in the search window and then select a matching word from the auto complete list, a trailing whitespace is automatically appended which isn't stripped-off the word before starting a search. The possibly results in a failed search.

### Expected behavior
If you enter a part of keyword in the search window and then select a matching word from the auto complete list, a trailing whitespace should automatically be stripped-off the word before starting a search.

### Steps to reproduce
1. Open the Nextcloud app and enter a part of a search term
2. Select a matching word from the auto complete list and check the parsed value.
   A trailing whitespace is automatically added which isn't stripped-off from the
   search term before starting a search.
-> The search fails with an empty result.

### Environment data
Android version: 11.0
Device model: Samsung Galaxy S20
Stock or customized system: Stock ROM
Nextcloud app version: 3.16.0 RC2
Nextcloud server version: 20.0.9

**Comment by tobiasKaminsky** [comment]
@nextcloud/server-triage can we do this on server side?

**Comment by MorrisJobke** [comment]
> @nextcloud/server-triage can we do this on server side?

Let's assume that @blizzz 👍 emojis means "Yes" xD

**Comment by blizzz** [comment]
> > @nextcloud/server-triage can we do this on server side?
> 
> Let's assume that @blizzz +1 emojis means "Yes" xD

My gut feeling is that the solution is a `trim()` away. #goodFirstIssue

**Comment by j-ed** [comment]
Due to the fact that no reference to a solution issue has been give, I assume that the problem hasn't been fixed yet.

**Comment by joshtrichards** [comment]
Closing in favor of nextcloud/server#33653 since this is a server matter
